app/src/main/java/com/twilio/video/app/sdk/ConnectOptionsFactory.kt

```diff
@@ -122,6 +104,26 @@ class ConnectOptionsFactory(
                 Preferences.MAX_AUDIO_BITRATE,
                 Preferences.MAX_AUDIO_BITRATE_DEFAULT)
 
+        val videoEncodingMode = sharedPreferences.get(Preferences.VIDEO_ENCODING_MODE,
+                Preferences.VIDEO_ENCODING_MODE_DEFAULT).let {
+            getVideoEncodingMode(it)
+        }
+
+        if (videoEncodingMode == VideoEncodingMode.AUTO) {
+            return createConnectOptions(token) {
+                roomName(roomName)
+                enableInsights(enableInsights)
+                enableAutomaticSubscription(enableAutomaticTrackSubscription)
+                enableDominantSpeaker(enableDominantSpeaker)
+                enableNetworkQuality(isNetworkQualityEnabled)
+                networkQualityConfiguration(configuration)
+                bandwidthProfile(bandwidthProfileOptions)
+                encodingParameters(EncodingParameters(maxAudioBitrate, 0))
+                preferAudioCodecs(listOf(preferredAudioCodec))
+                videoEncodingMode(videoEncodingMode)
+            }
```

Comment:
I had to branch off with a separate `createConnectOptions` call for enabling auto video encoding mode due to the non-nullable arguments of the `preferVideoCodecs` API call.